mandible/metadata_mapper/storage/storage.py

```diff
@@ -54,14 +81,20 @@ class FilteredStorage(Storage, register=False):
     """A storage which matches a set of filters on the context's files and
     returns data from the matching file.
     """
-    # Begin class definition
     filters: Dict[str, Any] = field(default_factory=dict)
 
     def __post_init__(self):
-        self._compiled_filters = {
-            k: re.compile(v) if isinstance(v, str) else v
-            for k, v in self.filters.items()
-        }
+        self._compiled_filters_cache: Optional[Dict[str, Any]] = None
+
+    @property
+    def _compiled_filters(self) -> Dict[str, Any]:
+        if self._compiled_filters_cache is None:
+            self._compiled_filters_cache = {
+                k: re.compile(v) if isinstance(v, str) else v
+                for k, v in self.filters.items()
```

Comment:
You can use `key` and `value` as well, but to me it seems unnecessary since this is a very common pattern that you will see all over python code. Similar to `for i in range(...)` you'll see `for k, v in some_dict.items()`.